File: PHP/ExibindoNumerosParesTest.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("ExibindoNumerosPares.php");

class ExibindoNumerosParesTest extends TestCase {

    public function testExibindoNumerosPares() {
        $obj = new ExibindoNumerosPares();
        //Test 1
        $want = [2, 4, 6];
        $got = $obj->countEve( 6 );
        $this->assertEquals($want, $got);
        //Test 2
        $want = [2, 4, 6, 8, 10, 12];
        $got = $obj->countEve( 13 );
        $this->assertEquals($want, $got);
    }

}

File: GO/ExibindoNumerosPares_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import (
	"reflect"
	"testing"
)

func TestExibindoNumerosPares(t *testing.T) {
	tables := []struct {
		N    int
		want []int
	}{
		{6, []int{2, 4, 6}},
		{13, []int{2, 4, 6, 8, 10, 12}},
	}

	for _, table := range tables {
		got := countEve(table.N)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %v but the return was this: %v", table.want, got)
		}
	}
}
